sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/healthcare/HttpHealthcareApiClient.java

```diff
@@ -0,0 +1,572 @@
+    return client.projects().locations().datasets().hl7V2Stores().get(storeName).execute();
+  }
+
+  @Override
+  public IngestMessageResponse ingestHL7v2Message(String hl7v2Store, Message msg)
+      throws IOException {
+    IngestMessageRequest ingestMessageRequest = new IngestMessageRequest();
+    ingestMessageRequest.setMessage(msg);
+    return client
+        .projects()
+        .locations()
+        .datasets()
+        .hl7V2Stores()
+        .messages()
+        .ingest(hl7v2Store, ingestMessageRequest)
+        .execute();
+  }
+
+  @Override
+  public HttpBody fhirSearch(String fhirStore, SearchResourcesRequest query) throws IOException {
+    return client
+        .projects()
+        .locations()
+        .datasets()
+        .fhirStores()
+        .fhir()
+        .search(fhirStore, query)
+        .execute();
+  }
+
+  @Override
+  public Message createHL7v2Message(String hl7v2Store, Message msg) throws IOException {
+    CreateMessageRequest createMessageRequest = new CreateMessageRequest();
+    createMessageRequest.setMessage(msg);
+    return client
+        .projects()
+        .locations()
+        .datasets()
+        .hl7V2Stores()
+        .messages()
+        .create(hl7v2Store, createMessageRequest)
+        .execute();
+  }
+
+  @Override
+  public HttpBody createFhirResource(String fhirStore, String type, HttpBody body)
+      throws IOException {
+    return client
+        .projects()
+        .locations()
+        .datasets()
+        .fhirStores()
+        .fhir()
+        .create(fhirStore, type, body)
+        .execute();
+  }
+
+  @Override
+  public Operation importFhirResource(
+      String fhirStore, String gcsSourcePath, @Nullable String contentStructure)
+      throws IOException {
+    GoogleCloudHealthcareV1beta1FhirRestGcsSource gcsSrc =
+        new GoogleCloudHealthcareV1beta1FhirRestGcsSource();
+
+    gcsSrc.setUri(gcsSourcePath);
+    ImportResourcesRequest importRequest = new ImportResourcesRequest();
+    importRequest.setGcsSource(gcsSrc).setContentStructure(contentStructure);
+    return client
+        .projects()
+        .locations()
+        .datasets()
+        .fhirStores()
+        .healthcareImport(fhirStore, importRequest)
+        .execute();
+  }
+
+  @Override
+  public Operation pollOperation(Operation operation, Long sleepMs)
+      throws InterruptedException, IOException {
+    LOG.debug(String.format("started opertation %s. polling until complete.", operation.getName()));
+    while (operation.getDone() == null || !operation.getDone()) {
+      // Update the status of the operation with another request.
+      Thread.sleep(sleepMs); // Pause between requests.
+      operation =
+          client.projects().locations().datasets().operations().get(operation.getName()).execute();
+    }
+    return operation;
+  }
+
+  @Override
+  public HttpBody executeFhirBundle(String fhirStore, HttpBody bundle) throws IOException {
```

Comment:
@lastomato @yeweidaniel @ygupta89
This is super awkward but the `executeBundle` method of client lib seems broken always giving errors about unparsable resource based on my testing (CDAP plugins also written similar to this).

Any ETA on fix for that?
Does this workaround seem acceptable til then?

Reply:
There is no ETA for a fix, and it is out of our control. It is acceptable to me :)